File: Array va String /PolindromSo'zlar.cs
bool is_palindrome(string word)
{
    var low = 0;
    var high = word.Length-1;
    
    while(low < high)
    {
        if(word[low] != word[high]) return false;

        low++;
        high--;      
    }
    return true;
    
}
System.Console.WriteLine(is_palindrome("aziza"));
 

File: Array va String /PolindromSo'zlar.py
"""
Berilgan so'z palindrom ekanligiga tekshiring.
Misol uchun:
  is_palindrome("kiyik") = True
  is_palindrome("aziza") = True
  is_palindrome("anora") = False
"""

def is_palindrome(word: str) -> bool:
    """
    Kodni bu yerda yozing.
    """
    # O(n) xotira, O(2n) = O(n)
    # return word = word[::-1]
    
    low, high = 0, len(word) -1
    
    while low < high:
        if word[low] != word[high]:
            return False
        low += 1
        high -= 1
    # O(n/2) - vaqt, O(1) - xotira
    return True